src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -44,11 +44,19 @@ extern "C" {
 
 // This is a basic implementation of the Halide runtime for Hexagon.
 void halide_print(void *user_context, const char *str) {
-    log_printf("%s", str);
+    if (str) {
+        log_printf("%s", str);
+    }
 }
 
 void halide_error(void *user_context, const char *str) {
-    halide_print(user_context, str);
+    if (!str) {
+        log_printf("Unknown error\n");
+    } else if (*str && str[strlen(str) - 1] != '\n') {
```

Comment:
I think you want "*str == '\0' || str[strlen(str) - 1] != '\n'" here. That way an empty string will be newline terminated. I realize I am being overly pedantic, but this sort of thing does show up when one logs some string that happens to unexpectedly be empty and having the lines run together makes debugging just a tiny bit more weird. As much as I like weird things, I like to avoid surprises when debugging.